smetrics/src/main/scala/com/evolutiongaming/smetrics/syntax/MeasureDurationSyntax.scala

```diff
@@ -0,0 +1,77 @@
+package com.evolutiongaming.smetrics.syntax
+
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.{Monad, MonadError}
+import com.evolutiongaming.smetrics.MeasureDuration
+
+import scala.concurrent.duration.FiniteDuration
+
+
+trait MeasureDurationSyntax {
+  implicit def measureDurationSyntax[F[_], A](fa: F[A]): MeasureDurationOps[F, A] =
+    new MeasureDurationOps[F, A](fa)
+}
+
+final class MeasureDurationOps[F[_], A](private val fa: F[A]) extends AnyVal {
+
+  /**
+   * Calculate time spent on F execution and handle result using provided function.
+   *
+   * Example:
+   * {{{
+   * import com.evolutiongaming.smetrics.syntax.measureDuration._
+   * val handler: FiniteDuration => F[Unit] = latency => Sync[F].delay { println(s"Successful execution. Latency is $latency") }
+   * Sync[F]
+   *   .delay { toBeMeasuredSideEffectCall() }
+   *   .measured { handler }
+   * }}}
+   *
+   * @param handleF function to consume calculated duration
+   * @return measured source F[A]
+   */
+  def measured(
+    handleF: FiniteDuration => F[Unit]
+  )(implicit F: Monad[F], measureDuration: MeasureDuration[F]): F[A] =
+    for {
+      measure <- measureDuration.start
+      result <- fa
+      duration <- measure
+      _ <- handleF(duration)
```

Comment:
I wonder should this also be attempted, or we consider nothing failable must be passed?

Reply:
measureCase for failable (MonadError required), here is just measured Monad F and no failure case possible. MonadError passed and failed - nothing measured

Reply:
I mean what if `handleF` also can fail?

Reply:
the same as for example:
```scala
for {
  _ <- log.info("start process")
  result <- service.run(data)
  _ <- metrics.record(result)
  _ <- appevent.send(result)
} yield ()
```
assuming handleF is a metric function, F will fail from metrics and next steps
We does not take care of logic, we just make utility function to avoid
```scala
for {
md <- MeasureDuration[F].start
_ <- service.call()
latency <- md
_ <- metrics.record(latency)
```

if you want to be safe pass function with .handleErrorWith clause